lib/gocardless_pro/services/payments_service.rb

```diff
@@ -34,6 +36,7 @@ def create(options = {})
           # Response doesn't raise any errors until #body is called
           response.tap(&:body)
         rescue InvalidStateError => e
+          raise IdempotencyConflict if @api_service.raise_on_idempotency_conflict
```

Comment:
I think this should be:

```ruby
if e.idempotent_creation_conflict?
  raise IdempotencyConflict if @api_service.raise_on_idempotency_conflict
  return get(e.conflicting_resource_id)
end
```

Reply:
Right. And i saw that errors have it's own place, so it would be nice to move `IdempotencyConflict` to the rest. Can i help you in any way?